protocol/testing/e2e/gov/prices_test.go

```diff
@@ -0,0 +1,200 @@
+	pricestest "github.com/dydxprotocol/v4-chain/protocol/testutil/prices"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	perptypes "github.com/dydxprotocol/v4-chain/protocol/x/perpetuals/types"
+	pricestypes "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	GENESIS_MARKET_PARAM = pricestypes.MarketParam{
+		Id:                 0,
+		Pair:               "btc-adv4tnt",
+		Exponent:           -8,
+		MinExchanges:       2,
+		MinPriceChangePpm:  1_000,
+		ExchangeConfigJson: "{}",
+	}
+
+	MODIFIED_MARKET_PARAM = pricestypes.MarketParam{
+		Id:                 0,
+		Pair:               "eth-adv4tnt",
+		Exponent:           -8,
+		MinExchanges:       3,
+		MinPriceChangePpm:  2_002,
+		ExchangeConfigJson: `{"exchanges":[{"exchangeName":"Bitfinex","ticker":"tBTCUSD"}]}`,
+	}
+)
+
+// This tests `MsgUpdateMarketParam` in `x/prices`.
+func TestUpdateMarketParam(t *testing.T) {
+	tests := map[string]struct {
+		msg                       *pricestypes.MsgUpdateMarketParam
+		expectCheckTxFails        bool
+		expectSubmitProposalFails bool
+		expectedProposalStatus    govtypesv1.ProposalStatus
+	}{
+		"Success": {
+			msg: &pricestypes.MsgUpdateMarketParam{
+				Authority:   lib.GovModuleAddress.String(),
+				MarketParam: MODIFIED_MARKET_PARAM,
+			},
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_PASSED,
+		},
+		"Failure: market param does not exist": {
+			msg: &pricestypes.MsgUpdateMarketParam{
+				Authority: lib.GovModuleAddress.String(),
+				MarketParam: pricestypes.MarketParam{
+					Id:                 GENESIS_MARKET_PARAM.Id + 1, // id does not exist
+					Pair:               MODIFIED_MARKET_PARAM.Pair,
+					Exponent:           MODIFIED_MARKET_PARAM.Exponent,
+					MinExchanges:       MODIFIED_MARKET_PARAM.MinExchanges,
+					MinPriceChangePpm:  MODIFIED_MARKET_PARAM.MinPriceChangePpm,
+					ExchangeConfigJson: MODIFIED_MARKET_PARAM.ExchangeConfigJson,
+				},
+			},
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_FAILED,
+		},
+		"Failure: exponent is updated": {
+			msg: &pricestypes.MsgUpdateMarketParam{
+				Authority: lib.GovModuleAddress.String(),
+				MarketParam: pricestypes.MarketParam{
+					Id:                 GENESIS_MARKET_PARAM.Id,
+					Pair:               MODIFIED_MARKET_PARAM.Pair,
+					Exponent:           MODIFIED_MARKET_PARAM.Exponent + 1, // update to exponent is not permitted.
+					MinExchanges:       MODIFIED_MARKET_PARAM.MinExchanges,
+					MinPriceChangePpm:  MODIFIED_MARKET_PARAM.MinPriceChangePpm,
+					ExchangeConfigJson: MODIFIED_MARKET_PARAM.ExchangeConfigJson,
+				},
+			},
+			expectedProposalStatus: govtypesv1.ProposalStatus_PROPOSAL_STATUS_FAILED,
+		},
+		"Failure: empty pair": {
+			msg: &pricestypes.MsgUpdateMarketParam{
+				Authority: lib.GovModuleAddress.String(),
+				MarketParam: pricestypes.MarketParam{
+					Id:                 GENESIS_MARKET_PARAM.Id,
```

Comment:
Question: if you are using `GENESIS_MARKET_PARAM.Id` instead of `MODIFIED_MARKET_PARAM.Id` here (presumably to show the id doesn't change,) does it also make sense to do the same thing with the exponent field? It seems like the treatment of these two fields is inconsistent.

Reply:
good call. modified to `MODIFIED_MARKET_PARAM.Id` to be consistent